tools/pickles_to_json.py

```diff
@@ -0,0 +1,534 @@
+    )
+    convert_parser.add_argument(
+        "--keys-file",
+        required=False,
+        help="Input file containing keys to perform the action on. One key per line.",
+    )
+    convert_parser.add_argument(
+        "--deprecated-keys-output",
+        required=False,
+        help="Output file to write deprecated keys to. These are keys associated with jobs not present in Tron. One key per line.",
+    )
+    convert_parser.add_argument(
+        "--failed-keys-output",
+        required=False,
+        help="Output file to write keys that failed to convert. One key per line.",
+    )
+    convert_parser.add_argument(
+        "--all",
+        action="store_true",
+        help="Apply the action to all keys in the table.",
+    )
+    convert_parser.add_argument(
+        "--dry-run",
+        action="store_true",
+        help="Simulate the conversion without making any changes to the DynamoDB table",
+    )
+
+    dump_pickle_parser = subparsers.add_parser("dump-pickle", help="Load and print the pickles for specified keys.")
+    dump_pickle_parser.add_argument(
+        "--keys",
+        nargs="+",
+        required=False,
+        help="Specific key(s) to perform the action on.",
+    )
+    dump_pickle_parser.add_argument(
+        "--keys-file",
+        required=False,
+        help="Input file containing keys to perform the action on. One key per line.",
+    )
+    dump_pickle_parser.add_argument(
+        "--all",
+        action="store_true",
+        help="Apply the action to all keys in the table.",
+    )
+
+    dump_json_parser = subparsers.add_parser("dump-json", help="Load and print JSON data for specified keys.")
+    dump_json_parser.add_argument(
+        "--keys",
+        nargs="+",
+        required=False,
+        help="Specific key(s) to perform the action on.",
+    )
+    dump_json_parser.add_argument(
+        "--keys-file",
+        required=False,
+        help="Input file containing keys to perform the action on. One key per line.",
+    )
+    dump_json_parser.add_argument(
+        "--all",
+        action="store_true",
+        help="Apply the action to all keys in the table.",
+    )
+
+    delete_keys_parser = subparsers.add_parser("delete-keys", help="Delete the specified keys from the DynamoDB table.")
+    delete_keys_parser.add_argument(
+        "--keys",
+        nargs="+",
+        required=False,
+        help="Specific key(s) to perform the action on.",
+    )
+    delete_keys_parser.add_argument(
+        "--keys-file",
+        required=False,
+        help="Input file containing keys to perform the action on. One key per line.",
+    )
+    delete_keys_parser.add_argument(
+        "--all",
+        action="store_true",
+        help="Apply the action to all keys in the table.",
+    )
+    delete_keys_parser.add_argument(
+        "--dry-run",
+        action="store_true",
+        help="Simulate the deletion without making any changes to the DynamoDB table",
+    )
```

Comment:
i'm personally a fan of dry-run by default, but that's just me :p

Reply:
Good call.

I took the easier (lazy) path here with an --execute flag and just passed `not args.execute` to the existing dry_run logic. I am down to rewrite the conversion functions a bit to avoid the negative logic, but only if you comment that 😛